src/libraries/System.Private.CoreLib/src/System/Convert.cs

```diff
@@ -2336,7 +2339,24 @@ public static string ToBase64String(ReadOnlySpan<byte> bytes, Base64FormattingOp
             }
 
             bool insertLineBreaks = (options == Base64FormattingOptions.InsertLineBreaks);
-            string result = string.FastAllocateString(ToBase64_CalculateAndValidateOutputLength(bytes.Length, insertLineBreaks));
+            int outputLength = ToBase64_CalculateAndValidateOutputLength(bytes.Length, insertLineBreaks);
+
+            if (!insertLineBreaks && bytes.Length >= 64)
+            {
+                // For large inputs it's faster to allocate a temp buffer and call UTF8 version
+                // which is then extended to UTF8 via Latin1.GetString (base64 is always ASCI)
+                [MethodImpl(MethodImplOptions.NoInlining)]
+                static string ToBase64StringLargeInputs(ReadOnlySpan<byte> data, int outputLen)
+                {
+                    Span<byte> utf8buffer = outputLen <= 256 ? stackalloc byte[256] : new byte[outputLen];
```

Comment:
For case > 256 rent from array-pool instead of allocating?

Reply:
This code has to allocate any way so I didn't bother making it zero alloc. Array pool is slower to rent/return and there are security concerns whether we need to zero what we do base64 for after usage.

Reply:
In `Encoding.Latin1.GetString` it allocates the string -- unavoidable here.
But the intermediate allocation of the buffer could be eliminated.

Reply:
I agree with @gfoidl... just because we need to allocate a returned string doesn't mean we should be haphazard about allocation.  Also for larger arrays (which would be the case here) ArrayPool can be faster to rent/return.  And there are many places we don't bother clearing arrays prior to returning them; in actuality the majority case is we don't.

|     Method | Length |      Mean | Ratio | Allocated | Alloc Ratio |
|----------- |------- |----------:|------:|----------:|------------:|
|      Alloc |    256 |  12.47 ns |  1.00 |     280 B |        1.00 |
| RentReturn |    256 |  15.04 ns |  1.21 |         - |        0.00 |
|            |        |           |       |           |             |
|      Alloc |   1024 |  42.36 ns |  1.00 |    1048 B |        1.00 |
| RentReturn |   1024 |  18.32 ns |  0.43 |         - |        0.00 |
|            |        |           |       |           |             |
|      Alloc |   4096 | 165.17 ns |  1.00 |    4120 B |        1.00 |
| RentReturn |   4096 |  15.37 ns |  0.09 |         - |        0.00 |

```C#
[Params(256, 1024, 4096)]
public int Length { get; set; }

[Benchmark(Baseline = true)]
public byte[] Alloc() => new byte[Length];

[Benchmark]
public byte[] RentReturn()
{
    byte[] arr = ArrayPool<byte>.Shared.Rent(Length);
    ArrayPool<byte>.Shared.Return(arr);
    return arr;
}
```